### PR TITLE
added constant tests

### PR DESCRIPTION
**Pull request by sakchal**
This PR introduces a suite of unit tests for constant.py located within the create_array directory. The primary functions in this file facilitate the creation of constant arrays, filled uniformly with a specified value. 

Correct Array Shape: Ensures that arrays are instantiated with the intended dimensions.
Data Type Accuracy: Validates that the resulting arrays strictly adhere to the specified data types.
Error Handling: Assesses the functions' error handling, specifically dealing with the shape of the array

### CODE REVIEW

**Review by roaffix — APPROVED** [review]
Short remark on imports: you could avoid calling methods from files and simply use `import arrayfire_wrapper.lib as wrapper`.

Apart from the minor comment, PR looks good to me. GJ!

**Comment by roaffix** [comment]
Tests are failing on code check. Consider running locally `make pre-commit` or explicitly `black --check .` to find the issues.
OR
you can autofix the issues by running `black .`

File: tests/test_constants.py [hunk 1]
```diff
@@ -37,33 +29,32 @@ def test_constant_shape(shape: tuple) -> None:
...
-)
-def test_constant_complex_shape(shape: tuple) -> None:
-    """Test if constant_complex creates an array with the correct shape."""
-    dtype = dtypes.c32
-    rand_array = randu((1, 1), dtype)
-    number = manage_array.get_scalar(rand_array, dtype)
 
-    if isinstance(number, (int, float, complex)):
-        result = constant_complex(number, shape, dtype)
+# @pytest.mark.parametrize(
```
Comment (roaffix) [inline tests/test_constants.py:37]: Why did you comment this part of the code?

File: tests/test_constants.py [hunk 2]
```diff
@@ -118,40 +109,40 @@ def test_constant_shape_invalid() -> None:
...
+        wrapper.constant(number, invalid_shape, dtype)
 
 
-def test_constant_complex_shape_invalid() -> None:
-    """Test if constant_complex handles a shape with greater than 4 dimensions"""
-    with pytest.raises(TypeError):
-        dtype = dtypes.c32
-        rand_array = randu((1, 1), dtype)
-        number = manage_array.get_scalar(rand_array, dtype)
+# def test_constant_complex_shape_invalid() -> None:
```
Comment (roaffix) [inline tests/test_constants.py:115]: aaaand this one?

File: tests/test_constants.py [hunk 3]
```diff
@@ -160,55 +151,55 @@ def test_constant_ulong_shape_invalid() -> None:
...
 
-@pytest.mark.parametrize(
-    "dtype_index",
-    [i for i in range(13)],
-)
-def test_constant_complex_dtype(dtype_index: int) -> None:
-    """Test if constant_complex creates an array with the correct dtype."""
-    if dtype_index not in [1, 3] or (dtype_index == 3 and not get_dbl_support()):
-        pytest.skip()
+# @pytest.mark.parametrize(
```
Comment (roaffix) [inline tests/test_constants.py:169]: and this?

File: tests/test_constants.py (hunk 1 above)
Comment (sakchal) [reply tests/test_constants.py]: The get_scalar method for complex arrays wasn't working, but I think I made a fix. Let me know if this is fine. I also uncommented all the previously commented methods.

**Comment by sakchal** [comment]
The code check is failing due to some mypy errors occurring on the backend.py file. I haven't touched that file, so if you want me to go in and fix those errors, I can do that as well.

**Pull request merged**